### DISCUSSION TITLE
[SOLVED] Selective cleanup

### DISCUSSION THREAD

**Discussion by slalaurette** [martinrotter/rssguard#1027] [Ideas]
Quite often I want to preserve old articles from some feeds but only delete articles older than X days from others. It would be nice if RSS Guard gave me an option to select the folders/feeds to include in a cleanup. At the moment I believe the cleanup applies to all folders and feeds. 

This could be implemented in two ways: 

1) In the cleanup dialog, letting me select the folders/feeds to apply the cleanup to (it can be "All" selected by default, or it could remember my last choice). This is the QuiteRSS way. 

2) In the right-click menu for a folder or feed, including a "Cleanup" menu item that only performs a cleanup to that folder/feed.

**Comment by slalaurette** [comment]
Bumping this because I'm drowning in old articles. ツ

No, but seriously, now that we have multiple selection for feeds, maybe it would be easier to have this? (Or just a context menu option for individual folders.)

**Reply by martinrotter** [reply]
OK, lets discuss this. Do you want "cleanup" to PURGE articles completely from DB (like "DB cleanup" dialog does) or just move them to recycle bin (like "cleanup selected items" menu item does)?

**Reply by slalaurette** [reply]
The idea would be that it would perform the exact same function as now (purge forever), but that you could apply it to selected folders/feeds. 

My use case is that I have several folders for news sites that I don't care about after a few days, but I also have some folders for blogs, etc. whose posts I don't want to lose until I've actually read them.

**Reply by slalaurette** [reply]
I've just discovered that if you select the "Clean selected items" menu item, and when the dialog appears, you choose "No", all the items are deleted anyway!

**Reply by martinrotter** [reply]
Kind of solved by https://github.com/martinrotter/rssguard/issues/1270
